src/c/types/array.c

```diff
@@ -20,195 +20,189 @@
 
 #include <string.h>
 
+static void ucdr_array_to_buffer(ucdrBuffer* ub, const uint8_t* array, const uint32_t size, const uint32_t data_size);
+static void ucdr_buffer_to_array(ucdrBuffer* ub, uint8_t* array, const uint32_t size, const uint32_t data_size);
+
 // -------------------------------------------------------------------
 //                INTERNAL SERIALIZATION IMPLEMENTATION
 // -------------------------------------------------------------------
 
-bool ucdr_serialize_array_byte_1(ucdrBuffer* ub, const uint8_t* array, const uint32_t size)
+void ucdr_array_to_buffer(ucdrBuffer* ub, const uint8_t* array, const uint32_t size, const uint32_t data_size)
 {
-    uint32_t data_size = sizeof(uint8_t);
-    if(ucdr_check_buffer(ub, size))
+    if(ucdr_check_buffer_available_for(ub, size))
     {
         memcpy(ub->iterator, array, size);
+        ub->iterator += size;
+    }
+    else
+    {
+        uint32_t remaining_size = size;
+        uint32_t serialization_size;
+        while(0 < (serialization_size = ucdr_check_final_buffer_behavior_array(ub, remaining_size, data_size)))
+        {
+            memcpy(ub->iterator, array + (size - remaining_size), serialization_size);
+            remaining_size -= serialization_size;
+            ub->iterator += serialization_size;
+        }
 
+        if(!ub->error)
```

Comment:
Yes, the previous function call to `ucdr_check_final_buffer_behavior_array` can set the error state to true. It is necessary to check it before the `memcpy`, because the state of the ucdrBuffer can be wrong (core exception cound happen)